5/Dockerfile

```diff
@@ -19,11 +18,16 @@ ENV JAVA_HOME=/usr/lib/jvm/java-17-openjdk \
 
 WORKDIR /opt
 
-RUN set -eux; \
-    addgroup -S -g ${GID} scanner-cli; \
-    adduser -S -D -u ${UID} -G scanner-cli scanner-cli; \
-    apk add --no-cache --virtual build-dependencies wget unzip gnupg; \
-    apk add --no-cache git python3 py-pip bash shellcheck 'nodejs>12' openjdk17-jre curl musl-locales musl-locales-lang; \
+SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
+RUN groupadd --system --gid 1000 scanner-cli; \
```

Comment:
Is there any particular reason why we are using semi-colons (`;`) instead of `&&` for the `RUN` commands?

Although not explicit, it seems that using `&&` is preferred since any intermediate error will make the build process crash, and the image will not be built.

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
https://cloud.google.com/architecture/best-practices-for-building-containers

http://docs.projectatomic.io/container-best-practices/#_using_double_ampersands_vs_semi_colons (outdated, but relevant)

Reply:
Ah! I missed that! Yes, it seems to achieve the same behavior!